potted plants; our trimmable friends

gy1ta23 opened the pull request:
🆑 gy1ta23, Taza
tweak: smaller potted plants are now objects. you can pick them up. rejoice
rscadd: small potted ferns are trimmable with wirecutters. the bridge now has a fern
/🆑

_Taza did the trimming code and sanitization for this._

Potted plants. My scourge. My bane. My ruination. These god-damnable little things, staring at me from their cathedras upon my desk. They know if they fall I have no chance of returning them to their proper place, no recourse. Before the smooth white polymer shell I stand a frail and broken man. My mind whimpers at their sight. I think I no longer possess the proper lucidity today to truly express what they have done to me, not anymore. How can a broken mind self-diagnose?

I do this for you, not for myself. I think I am, as I type this, beyond saving. The text on my monitor degenerates into the ramblings of a madman as my fingers move upon the keys. In these lines of code, I defenestrate these entities of unspeakable origin from their posts above us. They will go where we please, now, carryable as one desires (within the bounds of a large object.) Such power is not to be held lightly, but in these darkened times it is the only course of action available.

Trim them. Those terrible ferns hold things beyond words, beyond our coil. You **must** trim them. Wirecutters (or any object derived from such) will have to do, given the lack of scissors. Too much is better than too little. The Torch must be saved, and I no longer hold the capacity. I have enclosed further visual instructions, if such is necessary. Heed my warnings or find yourself knelt beside me.

![hedgetrim](https://user-images.githubusercontent.com/44277885/210124339-a4b3421f-b9f7-4caf-a0c7-22f01aaa95e2.gif)

My time runs short. Remember these words.

gy1ta23 commented:
trying to fix something, do not merge this

gy1ta23 commented:
My thoughts ebb away into the pelagic depths of an inscrutable sea. My figure is dotted with stains of carmine. I am not long for this world, body and mind. This work is complete, the one act I still yet claim dominion over. Take it, free of check failures, and live where I no longer can.

SierraKomodo commented on code/game/objects/items/flora.dm:
```diff
@@ -0,0 +1,36 @@
+	icon_state = "plant-27"
+	item_state = "plant-27"
+
+/obj/item/flora/pottedplantsmall/fern/attackby(obj/item/S as obj, mob/user as mob)
```
`as xyz` isn't necessary these days.

```suggestion
/obj/item/flora/pottedplantsmall/fern/attackby(obj/item/S, mob/user)
```

Tazaraki replied:
That's good to know! This will be fixed in the next update.

SierraKomodo commented on code/game/objects/items/flora.dm:
```diff
@@ -0,0 +1,36 @@
+	if (!isWirecutter(S))
+		return ..()
+	else
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		if (do_after(user, 6 SECONDS, src, DO_PUBLIC_UNIQUE))
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			to_chat (user, SPAN_NOTICE("You trim \the [src] with \the [S]. You probably should've used a pair of scissors."))
```
Public timers should use visible messages to provide context to anyone nearby about what the timer bar is for.

Tazaraki replied:
Noted. Will do that for future timers moving forward. Is there any particular standard for text styling that you recommend? For now, I am going to use SPAN_NOTICE.

SierraKomodo commented on code/game/objects/items/flora.dm:
```diff
@@ -0,0 +1,36 @@
+		if (do_after(user, 6 SECONDS, src, DO_PUBLIC_UNIQUE))
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			to_chat (user, SPAN_NOTICE("You trim \the [src] with \the [S]. You probably should've used a pair of scissors."))
+			new /obj/item/flora/pottedplantsmall/ferntrim(loc, src)
+			qdel(src)
```
I recommend making an icon state change instead of creating a new object and deleting the current one. It's a little cleaner that way and will prevent the apparent movement of the item on whatever it's sitting on due to randomized coordinates during initialization.

Best way to do that is something like this:

```dm
/obj/item/flora/pottedplantsmall/fern
    /// Boolean. Whether or not the fern has been trimmed.
    var/trimmed = FALSE

/obj/item/flora/pottedplantsmall/fern/attackby()
	// If statements and timer here
	update_icon()

/obj/item/flora/pottedplantsmall/ferm/on_update_icon()
    if (trimmed)
        icon_state = "plant-30"
    else
        icon_state = "plant-27"
```

Tazaraki replied:
Very cool. I've decided to completely obliterate the object /ferntrim from the code. There will now only be /fern and the icon_state, item_state, name, and desc will be handled by on_update_icon(). I also coded a grow() proc that lets the fern have its leaves back every 90 minutes. When that gets pushed, I'm eager for feedback on if how I did that was nonsensical or not.

gy1ta23 committed:
potted plants; our friends


potted plants; our friends


potted plants; our friends


potted plants; our friends


potted plants; fixing our friends (on their away maps)


potted plants; our (poorly indentated) friends


for the love of god stop failing checks


potted plants; our friends (dreammaker is not a friend)


fuck you, old-lab 2


green's replacement


potted plants; our friends (against whitespace)


there are no honorable discoveries in the darkest depths.
